### ISSUE TITLE
Rustdoc JSON: Duplicated ID's in lists

### ISSUE THREAD

**Issue by aDotInTheVoid** [rust-lang/rust#108236]
Looking at the output (of `tests/rustdoc-json/reexport/same_name_different_types.rs`), for some reason it's duplicating ID's in the output:

```json
  "index": {
    "0:0:1588": {
      "crate_id": 0,
      "docs": null,
      "id": "0:0:1588",
      "inner": {
        "is_crate": true,
        "is_stripped": false,
        "items": [
          "0:1:1582",
          "0:4-0:2:1584",
          "0:4-0:3:1584",
          "0:4-0:2:1584",
          "0:4-0:3:1584",
          "0:5-0:2:1584",
          "0:5-0:3:1584",
          "0:5-0:2:1584",
          "0:5-0:3:1584"
        ]
      },
      "kind": "module",
      "name": "same_name_different_types"
    },
```

I'm not really sure whats causing this, but idealy it wouldn't happen, although I'd be willing to merge as is, as long as an issue gets created to follow up.

(Also, jsondoclint should probably catch this)

_Originally posted by @aDotInTheVoid in https://github.com/rust-lang/rust/issues/107766#issuecomment-1434735864_

CC: #108163

**Comment by shogohida** [comment]
Hi @aDotInTheVoid, can I work on this? I searched a bit for the source code but I don't know where to start yet...

**Comment by aDotInTheVoid** [comment]
Your more than welcome to!

Convertion from clean to json for module's happens here, and it just passes of ID's coming from clean

https://github.com/rust-lang/rust/blob/e7eaed21d50d7bfb8d614d6ee7fcea940b39185d/src/librustdoc/json/conversions.rs#L302-L304

Clean makes it's modules here

https://github.com/rust-lang/rust/blob/e7eaed21d50d7bfb8d614d6ee7fcea940b39185d/src/librustdoc/clean/mod.rs#L51

I'd start by adding `debug!` calls to see whats going on here. See the rustc dev guide for info on how the debug macros work for rustc. Feel free to ask questions on zulip if you have any

**Comment by GuillaumeGomez** [comment]
It was fixed in #108349.
